addons/phantom_camera/scripts/phantom_camera/phantom_camera_3d.gd

```diff
@@ -534,11 +534,17 @@ func _ready():
 		if not Engine.is_editor_hint():
 			if not is_instance_valid(_follow_spring_arm):
 				_follow_spring_arm = SpringArm3D.new()
+				_follow_spring_arm.rotation = global_rotation
+				if is_instance_valid(follow_target):
+					_follow_spring_arm.position = _get_target_position_offset()
 				_follow_spring_arm.spring_length = spring_length
 				_follow_spring_arm.collision_mask = collision_mask
 				_follow_spring_arm.shape = shape
 				_follow_spring_arm.margin = margin
+				if not tween_on_load:
+					_has_tweened = true
 				get_parent().add_child.call_deferred(_follow_spring_arm)
+				reparent.call_deferred(_follow_spring_arm)
```

Comment:
The global changes above will trigger errors because of these two lines. The simple fix is to move the instantiation and reparenting calls to sit below the node creation `_follow_spring_arm = SpringArm3D.new()` on like 536.

Followed then by a node ready checker `if not _follow_spring_arm.is_node_ready(): await _follow_spring_arm.ready`.

tldr; the first couple of lines looking like:

```gdscript
if not is_instance_valid(_follow_spring_arm):
  _follow_spring_arm = SpringArm3D.new()
  get_tree().root.add_child.call_deferred(_follow_spring_arm)
  reparent.call_deferred(_follow_spring_arm)

  if not _follow_spring_arm.is_node_ready(): await _follow_spring_arm.ready

  # The rest

```

Reply:
I don't think it's good to have an `await` inside `_ready`. `global_transform == transform` also if the node is top_level, which is implied if you parent it to the root. We could just flag the SpringArm3D as top level, since that's the desired behavior anyways.

Reply:
In addition, adding the await would mean the user work around from #193 would need to also wait for `ready`.